### ISSUE TITLE
Naming of "contig" module and type

### ISSUE THREAD

**Issue by holtgrewe** [rust-bio/rust-bio-types#10]
I think the `contig` module and `Contig` struct have a naming issue. From my understanding, "contig" is a well-established term from genome assembly that describes a contiguously assembled sequence. This is, e.g., supported by the `contig` header in VCF... Also, I believe `genome::AbstractInterval::contig` shares the description.

What about renaming this as "linear"?

**Comment by dlaehnemann** [comment]
I agree. I stumbled upon this naming during the `rust-bio` Docathon in some code using the `Contig`. What is described as a `contig` in `contig.rs` is rather an interval on some reference sequence (and this reference sequence could more aptly be referred to as a contig...). The `Contig` `struct` does not include the sequence itself.

So it should definitely get a clearer name. What about `stranded_contig_interval`? A bit long, but it captures all the properties. Or maybe just `contig_interval`?
